plugin/src/main/java/io/github/rysefoxx/pagination/InventoryContents.java

```diff
@@ -2702,6 +2863,24 @@ public boolean update(@NotNull List<Integer> slots,
         return updated.get() >= slots.size();
     }
 
+    /**
+     * Update multiple items at once, with a new IntelligentItem.
+     *
+     * @param slots     The slots
+     * @param itemStack The new IntelligentItems that should be displayed.
+     * @return true if all items were updated, false if not.
+     * @throws IllegalArgumentException if slot greater than 53 or slot greater than inventory size
+     */
+    public boolean update(@NotNull List<Integer> slots,
+                          @NotNull IntelligentItem itemStack) throws IllegalArgumentException {
+        AtomicInteger updated = new AtomicInteger();
+        slots.forEach(integer -> {
+            update(integer, itemStack);
+            updated.getAndIncrement();
+        });
```

Comment:
Maybe make an if query around the update method before you increase the value. (I know I didn't do that either, I changed it in the meantime).